### PR TITLE
Add condition type to fix parsing order

### PR DESCRIPTION
**Pull request by TheLimeGlass**
### Description
Add condition type to fix parsing order. This is exactly the same as ExpressionType, shouldn't be a huge impact at all.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** https://github.com/SkriptLang/Skript/issues/5536, https://github.com/SkriptLang/Skript/issues/5804

this probably fixes other issues too.

### CODE REVIEW

**Review by Moderocky — APPROVED** [review]
Looks good, I'd like to make sure this isn't going to (somehow) break a bunch of scripts by changing the parsing orders.

**Comment by TheLimeGlass** [comment]
> Looks good, I'd like to make sure this isn't going to (somehow) break a bunch of scripts by changing the parsing orders.

2.8 betas will find out, I didn't find any from the tests, but I got the proper conditions to work, so that's a plus.

**Comment by TheLimeGlass** [comment]
Tests failing are unrelated. Currently failing outside this pull request.

**Review by sovdeeth — APPROVED** [review]

**Review by Moderocky — CHANGES_REQUESTED** [review]
Conflicts need fixing.

File: src/test/skript/tests/regressions/5804-is-burning.sk [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (APickledWalrus) [inline src/test/skript/tests/regressions/5804-is-burning.sk:1]:
```suggestion
```

**Review by APickledWalrus — APPROVED** [review]
Implementation is fine. The idea of condition priorities exists in <https://github.com/SkriptLang/Skript/pull/6246> which may end up getting merged for 2.10 (meaning this PR may only need to set priorities for something like CondCompare)

**Comment by TheLimeGlass** [comment]
When I created this pull request, I wanna add that this fix actually turned out surprisingly well, it fixed so many issues, and it's very simple in the fact that it's just modifying the list order like Expressions with ExpressionType.

**Review by Moderocky — APPROVED** [review]

**Comment by APickledWalrus** [comment]
Oh yeah I think the idea is good. I'm fine with merging this.

**Pull request merged**